### ISSUE TITLE
Exceptions in EmptyUnityMessageAnalyser and MessageSignatureAnalyser

### ISSUE THREAD

**Issue by SilentSin** [microsoft/Microsoft.Unity.Analyzers#32]
#### Bug description

I'm getting some warnings in the Visual Studio Error List from analyser exceptions which say they are happening in my [UltEvents](https://assetstore.unity.com/packages/tools/gui/ultevents-111307) plugin (it is showing 17 of these warnings, but it looks like there are only two different ones being triggered in multiple places).

- Version of analyzers assembly: 4.3.3.0
- Analyzer rule: EmptyUnityMessageAnalyser and MessageSignatureAnalyser

##### First warning

```
Severity	Code	Description	Project	File	Line	Suppression State	Detail Description
Warning	AD0001	Analyzer 'Microsoft.Unity.Analyzers.EmptyUnityMessageAnalyzer' threw an exception of type 'System.NullReferenceException' with message 'Object reference not set to an instance of an object.'.	UltEvents		1	Active	Analyzer 'Microsoft.Unity.Analyzers.EmptyUnityMessageAnalyzer' threw the following exception:
'Exception occurred with following context:
Compilation: UltEvents
SyntaxTree: C:\Users\Kailas\Source\Repos\Games\Mage Wars\Assets\Plugins\UltEvents\Events\UltEventBase.cs
SyntaxNode: PersistentCall AddPersistentCall ... [MethodDeclarationSyntax]@[699..749) (20,8)-(20,58)

System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.Unity.Analyzers.EmptyUnityMessageAnalyzer.AnalyzeMethodDeclaration(SyntaxNodeAnalysisContext context)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.<>c__53`1.<ExecuteSyntaxNodeAction>b__53_0(ValueTuple`2 data)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.ExecuteAndCatchIfThrows_NoLock[TArg](DiagnosticAnalyzer analyzer, Action`1 analyze, TArg argument, Nullable`1 info)
```

##### Second Warning

```
Severity	Code	Description	Project	File	Line	Suppression State	Detail Description
Warning	AD0001	Analyzer 'Microsoft.Unity.Analyzers.MessageSignatureAnalyzer' threw an exception of type 'System.ArgumentException' with message 'An item with the same key has already been added.'.	UltEvents		1	Active	Analyzer 'Microsoft.Unity.Analyzers.MessageSignatureAnalyzer' threw the following exception:
'Exception occurred with following context:
Compilation: UltEvents
SyntaxTree: C:\Users\Kailas\Source\Repos\Games\Mage Wars\Assets\Plugins\UltEvents\Inspector\ObjectPicker.cs
SyntaxNode: internal sealed class ObjectPickerWindow ... [ClassDeclarationSyntax]@[11004..29499) (218,4)-(717,5)

System.ArgumentException: An item with the same key has already been added.
   at System.ThrowHelper.ThrowArgumentException(ExceptionResource resource)
   at System.Collections.Generic.Dictionary`2.Insert(TKey key, TValue value, Boolean add)
   at System.Linq.Enumerable.ToDictionary[TSource,TKey,TElement](IEnumerable`1 source, Func`2 keySelector, Func`2 elementSelector, IEqualityComparer`1 comparer)
   at Microsoft.Unity.Analyzers.MessageSignatureAnalyzer.AnalyzeClassDeclaration(SyntaxNodeAnalysisContext context)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.<>c__53`1.<ExecuteSyntaxNodeAction>b__53_0(ValueTuple`2 data)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.ExecuteAndCatchIfThrows_NoLock[TArg](DiagnosticAnalyzer analyzer, Action`1 analyze, TArg argument, Nullable`1 info)
```

#### To Reproduce
Steps or code to reproduce the behavior:
1. Create a Unity project (I'm using 2019.3.2f1) and import [UltEvents](https://assetstore.unity.com/packages/tools/gui/ultevents-111307).
2. Open `Assets\Plugins\UltEvents\Events\UltEventBase.cs` and `Assets\Plugins\UltEvents\Inspector\ObjectPicker.cs`.
3. Wait for the analysers to run I guess?
4. Check the Error List window.

**Comment by sailro** [comment]
The first bug regarding `EmptyUnityMessageAnalyzer` was -I think- fixed, it was related to https://github.com/microsoft/Microsoft.Unity.Analyzers/blob/master/src/Microsoft.Unity.Analyzers/EmptyUnityMessage.cs#L51. (we previously failed when inspecting an interface method)

The second bug regarding `MessageSignatureAnalyzer` was already fixed with https://github.com/microsoft/Microsoft.Unity.Analyzers/commit/97a47aec437548df8bbd6c4a5224122d97c7451d#diff-55ba75c53c3d24053272d25ba9f22b46. 

I'm going to try your repro to make sure everything is ok.

**Comment by sailro** [comment]
Confirmed, no issue with the master.

![image](https://user-images.githubusercontent.com/638167/75369374-306d2300-5878-11ea-9b11-fa58ef86d8c2.png)

Fixes will be available with `VS 2019 16.5`, `VS for Mac 8.5`.

Thank you for the report @SilentSin !
